assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -158,22 +143,17 @@ function LibraryProvider({ children }) {
       setPageCanvasPromise,
     ]
   );
-  const getTextSets = useCallback(async () => {
-    const trackTiming = getTimeTracker('load_text_sets');
-    setTextSets(await loadTextSets());
-    trackTiming();
-  }, []);
-
   useEffect(() => {
-    // track the rendered tabs
-    const previouslyRenderedTabs = { ...renderedTabs.current };
-    renderedTabs.current = { ...previouslyRenderedTabs, [tab]: true };
-
-    // if text pane hasn't been rendered until now fetch dynamically imported text sets
-    if (tab === TEXT.id && !previouslyRenderedTabs[tab]) {
+    async function getTextSets() {
+      const trackTiming = getTimeTracker('load_text_sets');
+      setTextSets(await loadTextSets());
+      trackTiming();
+    }
+    // if text sets have not been loaded but are needed fetch dynamically imported text sets
+    if (tab === TEXT.id && !Object.keys(textSets).length) {
```

Comment:
This could also be moved directly to the text tab now instead perhaps?

Reply:
Yeah, I wasn't sure why it was here in the first place, as it was always invoked the first time the tab was shown anyway, but I'm guessing there was a reason?

Reply:
Hm, not sure what could be the reason, perhaps access to the `previouslyRenderedTabs`? Looks like currently we can move it though, doesn't seem to have a reason to stay here anymore 🤷